examples/edge_update.py

```diff
@@ -0,0 +1,32 @@
+import os.path as osp
+
+from torch_geometric.nn import MessagePassing
+from torch_geometric.data import Data
+
+from torch_geometric.datasets import Planetoid
+import torch_geometric.transforms as T
+
+dataset = 'Cora'
+transform = T.Compose([
+    T.RandomNodeSplit(num_val=500, num_test=500),
+    T.TargetIndegree(),
+])
+path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', dataset)
+dataset = Planetoid(path, dataset, transform=transform)
+data = dataset[0]
+
+class EdgeMessagePassingExample(MessagePassing):
+
+    def forward(self, data: Data):
+        x = self.propagate(data.edge_index, edge_attr=data.edge_attr, x=data.x)
+        edge_attr = self.edges_propagate(data.edge_index, edge_attr=data.edge_attr, x=data.x)
```

Comment:
We do not really propagate information here. I would prefer the earlier naming of `update_edge` (or something similar).

Reply:
I liked update_edge too, but I hate `update_edges` for the propagate and `edge_updates` for the message function.. which seemed a bit confusing.